### PR TITLE
knot: upgrade to 2.5.6

### PR DESCRIPTION
**Pull request by salzmdan**
Signed-off-by: Daniel Salzman <daniel.salzman@nic.cz>

Maintainer: me
Compile tested: mpc85xx (8540) and mvebu (armv7-a), TurrisOS (OpenWRT 14.07)
Run tested: mvebu (armv7-a), TurrisOS (OpenWRT 14.07)

### CODE REVIEW

**Comment by salzmdan** [comment]
@pprindeville Please recheck.

**Comment by champtar** [comment]
@pprindeville if you don't have anymore remarks please merge

**Comment by salzmdan** [comment]
I must fix the problem with libatomic :hourglass_flowing_sand:

File: net/knot/Makefile [hunk 1]
```diff
@@ -40,75 +41,76 @@ define Package/knot/Default
...
-
-define Package/knot-libknot
-	$(call Package/knot-lib/Default)
-	TITLE+= library
-	DEPENDS=+libpthread +knot-libdnssec
+	TITLE+= common DNS and DNSSEC libraries
+	DEPENDS+=+libgnutls +jansson
 endef
 
 define Package/knot-libzscanner
```
Comment (pprindeville) [inline net/knot/Makefile]: This is a little confusing, there's a "libs" but this isn't part of it.

I liked it better when the previous library was called "knot-libdnssec"

File: net/knot/Makefile [hunk 2]
```diff
@@ -40,75 +41,76 @@ define Package/knot/Default
...
 	TITLE+= zone parser library
-	DEPENDS=+libpthread
 endef
 
 define Package/knot
 	$(call Package/knot/Default)
-	TITLE+= server
-	DEPENDS=+liburcu +libedit +knot-libdnssec +knot-libknot +knot-libzscanner
+	TITLE+= server with control utility
+	DEPENDS=+libedit +liburcu +knot-libs +knot-libzscanner
```
Comment (pprindeville) [inline net/knot/Makefile]: Travis is complaining:

```
Package knot is missing dependencies for the following libraries:

libatomic.so.1

make[2]: *** [/tmp/tmp.pVDHpKRJ6y/bin/packages/mips_24kc/packages/knot_2.5.4-1_mips_24kc.ipk] Error 1
```

so please add `+libatomic`

**Comment by salzmdan** [comment]
Please recheck :recycle:

**Comment by champtar** [comment]
ping @pprindeville

**Comment by ja-pa** [comment]
@pprindeville  This version also fix CVE-2017-11104 https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-11104 
Version 2.3.3  is still vulnerable.

**Pull request merged**